File: UI/zemljaslova_mobile/lib/models/recommendation.dart
import 'package:json_annotation/json_annotation.dart';
import 'book.dart';
import 'member.dart';

part 'recommendation.g.dart';

@JsonSerializable()
class Recommendation {
  final int id;
  final int memberId;
  final int bookId;
  final Book? book;
  final Member? member;

  Recommendation({
    required this.id,
    required this.memberId,
    required this.bookId,
    this.book,
    this.member,
  });

  factory Recommendation.fromJson(Map<String, dynamic> json) => 
      _$RecommendationFromJson(json);

  Map<String, dynamic> toJson() => _$RecommendationToJson(this);
}

File: ZemljaSlova.Model/Recommendation.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace ZemljaSlova.Model
{
    public partial class Recommendation
    {
        public int Id { get; set; }

        public int MemberId { get; set; }

        public int BookId { get; set; }

        public virtual Book Book { get; set; } = null!;

        public virtual Member Member { get; set; } = null!;
    }
}
